WIP Swarm/Consul

till opened the pull request:
For: pngmbh/issues#496

till committed:
Update: switch to full swarm mode

till committed:
Fix: only use basic auth, when enabled

till committed:
Update: run consul global (WIP!)

rwos commented:
re: traefik test: it's using `Host:traefik.traefik-swarm.docker.local` but molecule/default/playbook.yml has `traefik_docker_domain: 'traefik.docker.local'`, not `traefik-swarm.docker.local`

till committed:
Update: use "latest" 1.7

rwos commented:
the storeconfig doesn't seem to work on circleci, sometimes(?): from the traefik_init container:
```
2020/02/26 17:28:27 Error running traefik: Put http://consul:8500/v1/kv/traefik/accesslogsfile?flags=3304740253564472344&wait=30000ms: dial tcp 10.0.2.2:8500: connect: no route to host
```

rwos commented:
So, it's nothing wrong with consul, I think somehow the whole docker network stuff doesn't work - can't ping or connect to any container from any other one (but the hostnames all resolve).

No idea where in the layers of docker-swarm in docker in circleci it all goes wrong, but probably somewhere.

rwos committed:
debug: try using the "host" (circleci-machine's) docker

rwos committed:
Revert "debug: try using the "host" (circleci-machine's) docker"

This reverts commit a930efba41faa7de34db12d60156b401d17374ba.

rwos committed:
Chore: switch to hcloud for swarm tests

Docker-in-docker-in-docker is difficult to get working.

rwos commented:
now the `pip install docker` fails, for some reason. So much fun.

till commented:
Oh, I just remembered. I think we have to configure a network explicitly in Molecule. No idea. I remember wasting a few hours on that before.

rwos commented:
I switched this over to hetzner now, so that *should* work now. But haven't looked through the pip error messages yet. It's a bit weird that this doesn't work out of the box, because the same stuff works in other repos...

rwos committed:
Chore: remove system-level python packages prior to pip install

rwos commented:
aaaand it's back at the same error. Now that was useful 🙈

rwos committed:
Fix: set placement constraint to something working in molecule test

rwos committed:
Chore: hack around traefik startup issues

rwos commented on molecule/swarm/playbook.yml:
```diff
@@ -4,20 +4,48 @@
+        cmd: docker service update --force traefik_traefik
+    - name: Wait for traefik/consul to come up
+      wait_for:
+        port: 80
```
So this works. It's also horrible. From hacking around it seems one needs to do one request and then restart traefik before everything works properly. I have no idea why. Sometimes, just a restart is enough - but what never seems to work is just letting it all come up cleanly from scratch.

*Maybe* things would work better if traefik would actually wait until the consul is running and configured before it starts up, but not 100% sure if that is the core of the issue.

till committed:
Update: make consul version configurable

till committed:
Update: more debug

till committed:
Fix: prune services

till committed:
test consul (#16)

* Update: install consul

* Fix: tests

 - for context, the curl succeeds but because of -k there is a different exit code
 - turn off -v so we can compare output
 - add/leave some debug code for our future selves

rwos commented on templates/docker-compose.yml.j2:
```diff
@@ -5,101 +5,33 @@ services:
-      - "-bootstrap-expect=1"
-      - "-log-level=debug"
-      - "-client=0.0.0.0"
+      - "--configfile=/traefik.toml"
```
aahh 😄 👍

rwos reviewed (approved):
LGTM
